### ISSUE TITLE
[Edge-RM CH02] New Profile for Edge ( vRAN ,..)

### ISSUE THREAD

**Issue by ASawwaf** [cntt-n/CNTT#1569]
Dears, this issue for more decision about how can we cover the vRAN or/and any workload required the same requirements ( FPGA ) at the edge, we agreed during Edge meeting that we need to have a new profile so 

we have 2 options : 
1- Re-activate the Compute intensive 
2- Create a new profile

**Comment by kedmison** [comment]
Compute-optimized was intended as a 'better' version of basic; either faster vCores, or more vCores.  FPGAs are an entirely different computational paradigm, using entirely different binaries (FPGA bitstreams), rather than x86 instruction sets.

For this reason, I would recommend introducing this as a FPGA profile.  
This profile should also explore the required vCPU count, RAM, etc and also address how many of these FPGAs may be installed in the system, and how they are associated to workloads.

**Comment by kedmison** [comment]
RM meeting 2020-05-13: We are not opposed to a workload profile for FPGA, but need vendor-independent characterization of the  FPGA adapter card capabilities prior to defining the profile.

**Comment by ASawwaf** [comment]
thanks all for our discussion today, we will continue to cover this in edge meeting 2020-05-14
but it makes my life more complicated, i need big support from @petorre @trevgc if we can find some characterization, which I believe the FPGAs as Technologies by nature has independent characterization due to the Fact of the use case that it used for

**Comment by trevgc** [comment]
I was told that even today it is possible to design a very low capacity FlexRAN solution without FPGA. To be verified what this really means but I think highlights that we need to be careful about assuming a profile is dependent on any particular technology.

**Comment by TFredberg** [comment]
I agree with @trevgc that CNTT should not assume that it is impossible to run a specific workload without some special HW that might make the workload run more efficient or live up to hard characteristics requirements.

There is however a need in all cases to specify the characteristics needs from the workloads and the characteristics capabilities from the infrastructure resource instances in any deployed Cloud Infrastructure. These characteristics needs to be expressed in a way that make sense both for the workload and the resource provider. 

It think it would be very good if the involved subject matter expert around the 4 use cases could have a first suggestion about what characteristics matter and see if they could express them in a technology independent way, but still be useful for the resource provisioning when inserting an FPGA, GPU or SmartNIC PCIe card.

**Comment by lucisuciu** [comment]
Dear All, 

IMHO, I'll go for the creation of a new edge profile, and it should be techology-independent if possible... 

Remember that we aim at several edge use-cases, and we can have FPGAs, GPUs, eASICs, etc. out there :)

This is not easy, I know...

Thanks.

**Comment by Karimrabie** [comment]
> I agree with @trevgc that CNTT should not assume that it is impossible to run a specific workload without some special HW that might make the workload run more efficient or live up to hard characteristics requirements.
> 
> There is however a need in all cases to specify the characteristics needs from the workloads and the characteristics capabilities from the infrastructure resource instances in any deployed Cloud Infrastructure. These characteristics needs to be expressed in a way that make sense both for the workload and the resource provider.
> 
> It think it would be very good if the involved subject matter expert around the 4 use cases could have a first suggestion about what characteristics matter and see if they could express them in a technology independent way, but still be useful for the resource provisioning when inserting an FPGA, GPU or SmartNIC PCIe card.

I do agree. We need to define the characteristics of the use cases and map that to technology independent aspects. 

Let me group some initial characteristics of the use cases and post it for group review. 

I prefer creating a new profile for the edge but we need to document the use cases characteristics as an initial step.

**Comment by ASawwaf** [comment]
@kedmison @petorre @trevgc @rabi-abdel @Karimrabie @lucisuciu @TFredberg 
Thanks for the discussion, we agreed during last week Edge meeting that **we will create a new profile** 
and we have an action point 
1- @petorre , will define some content to cover 3 points in MoM, in terms of the abstraction required,...
2- It will not be called a FPGA profile as we are not sticking to certain Tech.
3- It can be a candidate to be in transaction technologies plan like SRIOV ( need discussion to see how can we cater it )

https://wiki.lfnetworking.org/display/LN/2020-05-14+-+%5BCNTT+EDGE%5D+-+Meeting+Agenda+and+Minutes

@petorre , i will great if you can have it by next meeting

**Comment by petorre** [comment]
Initial lines in:

Compared to other workloads and HW profiles:
Incremental requirements: Latency and Timing Accuracy, Edge-limited power envelope.
Acceleration for functionality like: Fronthaul IO and 5G Channel Coding (Forward Error Correction, FEC).
Acceleration implemented with: Programmable (like FPGA) or Fixed Function (for stable algorithms).

**Comment by ASawwaf** [comment]
@petorre , can you prepare some details for the next meeting

**Comment by ASawwaf** [comment]
As per our the last Edge meeting  ( 11 of June ), we will go to define a new profile called **"Edge Profile"**
https://wiki.lfnetworking.org/display/LN/2020-06-11+-+%5BCNTT+EDGE%5D+-+Meeting+Agenda+and+Minutes 

I will start the PR to cover it

**Comment by pgoyal01** [comment]
PR #1692 closed w/o merge. This issue will be addressed in next release.
